### ISSUE TITLE
st.waveform_display requires removed peak_classification; update demo

### ISSUE THREAD

**Issue by aladinggit** [XENONnT/straxen#48]
In #36, peak_classification was removed.  This seems to be used in `st.waveform_display` but I am not sure of the appropriate fix.

Additionally, the demos should be updated to reflect peak_classification missing.  For example: https://github.com/XENONnT/straxen/blob/master/notebooks/tutorials/strax_demo.ipynb

See:

`st.waveform_display('170204_2111', seconds_range=(0, 0.15))`

```---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-3f3d31936807> in <module>
----> 1 st.waveform_display('170204_2111', seconds_range=(0, 0.15))

/dali/lgrandi/strax/straxen/straxen/mini_analysis.py in wrapped_f(context, run_id, **kwargs)
     91             if len(requires):
     92                 deps_by_kind = strax.group_by_kind(
---> 93                     requires, context=context, require_time=False)
     94                 for dkind, dtypes in deps_by_kind.items():
     95                     if dkind in kwargs:

/dali/lgrandi/strax/strax/strax/utils.py in group_by_kind(dtypes, plugins, context, require_time)
    472         if context is None:
    473             raise RuntimeError("group_by_kind requires plugins or context")
--> 474         plugins = context._get_plugins(targets=dtypes, run_id='0')
    475 
    476     if require_time is None:

/dali/lgrandi/strax/strax/strax/context.py in _get_plugins(self, targets, run_id)
    413         plugins = collections.defaultdict(get_plugin)
    414         for t in targets:
--> 415             p = get_plugin(t)
    416             # This assignment is actually unnecessary due to defaultdict,
    417             # but just for clarity:

/dali/lgrandi/strax/strax/strax/context.py in get_plugin(d)
    359 
    360             if d not in self._plugin_class_registry:
--> 361                 raise KeyError(f"No plugin class registered that provides {d}")
    362 
    363             p = self._plugin_class_registry[d]()

KeyError: 'No plugin class registered that provides peak_classification'```

Written with @tunnell

**Comment by JelleAalbers** [comment]
Hi Hongyi, Thanks for bringing this up! The fix for `waveform_display` is already in master (#44), but indeed, the tutorial still has to be updated. `waveform_display` still doesn't work even with this fix; as #44 mentions there is some ZeroDivisionError in a range computation inside holoviews.

**Comment by tunnell** [comment]
Is this fixed deployed to an environment on Midway? We are hitting other issues related to this change.  What is the simplest example that should work if it's not:

```
import straxen
st = straxen.contexts.strax_workshop_dali()
run_id = '170206_1555'
st.plot_waveform(run_id, seconds_range=(50.27050, 50.27075))
````

which gives: `ValueError: Arrays to merge must have the same length, got lengths 1432, 1055`

**Comment by tunnell** [comment]
Are these minianalyses tested in continuous integration?  Just that they don't crash?

**Comment by JelleAalbers** [comment]
Thanks for reporting, I can reproduce. This is a different and new issue to do with the time selection in some runs (or maybe some parts of some runs). I could not get a waveform plot from 170206_1555, but it worked fine for 170607_0919 and 180215_1029 (which we usually test with). Then I tried another and my kernel blew up, probably out of memory. I suspect the time range to chunk range translation is incorrect.

Good idea to test the mini-analyses. Even with the few-seconds (or however long it is) of data we dared to ship, we can still do a non-crash test. Though that wouldn't have caught this bug.
